xbmc/addons/AddonRepos.cpp

```diff
@@ -270,3 +271,59 @@ bool CAddonRepos::GetLatestAddonVersion(const std::string& addonId, AddonPtr& re
 
   return true;
 }
+
+bool CAddonRepos::FindDependency(const std::string& dependsId,
+                                 const AddonPtr& parent,
+                                 AddonPtr& dependencyToInstall,
+                                 std::shared_ptr<CRepository>& repoForDep) const
+{
+  const AddonRepoUpdateMode updateMode =
+      CAddonSystemSettings::GetInstance().GetAddonRepoUpdateMode();
+
+  bool hasOfficialVersion =
+      GetLatestVersionByMap(dependsId, m_latestOfficialVersions, dependencyToInstall);
+
+  if (!hasOfficialVersion)
```

Comment:
I'm not sure this logic is correct. Regardless of whether or not there is an official version of the addon available we want the ANY REPO setting to be respected. What we are looking for here is if the version in any other repo is higher than the official one (if there is one). Correct?